api/v1alpha1/volumereplicationgroup_types.go

```diff
@@ -18,53 +18,38 @@ package v1alpha1
 
 import (
 	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	vrv1alpha1 "github.com/kube-storage/volume-replication-operator/api/v1alpha1"
 )
 
-// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
 // NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
 
-// VolumeTakeoverControlType -- takeover control when desiredCluster is different from affinedCluster
-type VolumeTakeoverControlType string
-
-// VolumeTakeoverControlType definitions
-const (
-	// Force promote the volume in a WAN DR setting
-	ForcePromote VolumeTakeoverControlType = "ForcePromote"
-)
-
-// VolumeReplicationGroupSpec defines the desired state of VolumeReplicationGroup
+// VolumeReplicationGroup (VRG) spec declares the desired replication class
+// and replication state of all the PVCs owned by a given application, which
+// are identified via the given PVC label selector.  For every PVC created
+// by the application, the VRG operator does the following:
+// 	- Create a VolumeReplication (VR) CR to enable storage level replication
+// 	  of volume data and set the volume state to primary, secondary, etc.
+//  - Take the corresponding PV metadata in Kubernetes etcd and deposit it in
```

Comment:
(future) The destination S3 store/location may need to be fed per VRG CR.

- A cluster may be in a DR relationship across different peer groups
- Each peer group hence may have a different S3 bucket(?) or target location
- URL should hence be configured per VRG
- BUT, access tokens to the S3 store would need to still come from secrets

Also, this is possibly another case where we need a notion of DRClusterPeers CRD to group DR clusters.

Reply:
I too was thinking on the same lines, but wasn't sure how AVR is going to provide it, unless the S3 url is an AVR spec field.  We should discuss the connection you're making with DRClusterPeers; will AVR create it or OCS?

Reply:
I created a new issue https://github.com/RamenDR/ramen/issues/19

Reply:
I think we always had in mind that each VRG will have the S3 endpoint (per the original diagram).  I thought that will come later.  So I think we do agree that each VRG will be fed the S3 endpoint

Reply:
Issue open to discuss further, marking this as resolved.